recipe/build-arrow.bat

```diff
@@ -10,26 +10,76 @@ if "%cuda_compiler_version%"=="None" (
     set "EXTRA_CMAKE_ARGS=-DARROW_CUDA=ON"
 )
 
+:: Set CMAKE components based on package
+if [%PKG_NAME%] == [libarrow-acero] (
+    set "ARROW_PKG_CMAKE_ARGS=-DARROW_ACERO=ON ^
+                              -DARROW_COMPUTE:BOOL=ON ^
+                              -DARROW_DATASET:BOOL=OFF ^
+                              -DARROW_FLIGHT:BOOL=OFF ^
+                              -DARROW_FLIGHT_REQUIRE_TLSCREDENTIALSOPTIONS:BOOL=OFF ^
+                              -DARROW_FLIGHT_SQL:BOOL=OFF ^
+                              -DARROW_GANDIVA:BOOL=OFF ^
+                              -DARROW_SUBSTRAIT:BOOL=OFF"
+) else if [%PKG_NAME%] == [libarrow-dataset] (
+    set "ARROW_PKG_CMAKE_ARGS=-DARROW_ACERO=ON ^
```

Comment:
I think it would be much easier to read if we initially all defaults to OFF and then override only the ones relevant per package to true. Something like:
```suggestion
set "ARROW_ACERO=OFF"
set "ARROW_COMPUTE=OFF"
set "ARROW_DATASET=OFF"
set "ARROW_FLIGHT=OFF"
set "ARROW_FLIGHT_TLS=OFF"
set "ARROW_FLIGHT_SQL=OFF"
set "ARROW_GANDIVAT=OFF"
set "ARROW_SUBSTRAIT=OFF"
:: Set CMAKE components based on package
if [%PKG_NAME%] == [libarrow-acero] (
    set "ARROW_ACERO=ON"
    set "ARROW_COMPUTE=ON"
) else if [%PKG_NAME%] == [libarrow-dataset] (
    set "ARROW_ACERO=ON"
    set "ARROW_COMPUTE=ON"
    set "ARROW_DATASET=ON"
) ...
```